nahusenayElias/Team_4
# DRD-90: Add Mautic form to drupal and fetch it to frontend

## Related ticket 
[DRD-90](https://edu-team4-react24k.atlassian.net/jira/software/projects/DRD/boards/1?selectedIssue=DRD-90)
## In this PR:
- Added a Mautic form block in Drupal
- Removed the Mautic tracking script from backend
- Fetched Mautic form to frontend using the form id via JSON:API
NOTE! I did not make changes to the mautic paragraph module this time so it should work as it is and therefore the files are not included in this PR. Also, the form is not displayed in Drupal right now (I tried adding the form block to the content section and that broke the whole site for me). But it should now be working correctly on the frontend. 
## Testing instructions
- checkout branch (DRD-90-2)
- lando db-import DRD-90-new.gz
- lando composer install
- lando drush cim
- lando drush cr
- Drupal > Configuration > Web Services > Mautic integration: check that your credentials are correct and connection to Mautic is working. 
- Drupal > Content > Blocks: check that you can see the mautic form block in the list
![image](https://github.com/user-attachments/assets/af358ccb-dede-449d-81c4-ce4f4a70ea71)
- Frontend: Check the Contact page that form has loaded and is rendered to the frontend correctly.
- Submit the form and check that the form data is coming to Mautic successfully. 
- Check that console has no errors. 
- Try navigating to other pages after viewing the contact page (previously this broke the site but should now be working normally)

Let me know if there are any issues! I might have changed something in the Mautic paragraph module without remembering it so I can also include the module files in the PR if needed. 

[DRD-90]: https://edu-team4-react24k.atlassian.net/browse/DRD-90?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
